### ISSUE TITLE
args[module] flags variables violation

### ISSUE THREAD

**Issue by timo-fc** [ansible/ansible-lint#2866]
##### Summary

The args[module] flags variables violation

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.10.2 using ansible 2.14.1
```

##### STEPS TO REPRODUCE

Here one example:

```
- name: "Add or modify pam limits"
  community.general.pam_limits:
    domain: "{{ item.domain }}"
    limit_type: "{{ item.limit_type }}"
    limit_item: "{{ item.limit_item }}"
    value: "{{ item.value }}"
  with_items:
    - { domain: "root", limit_type: "soft", limit_item: "nofile", value: "4096" }
    - { domain: "root", limit_type: "hard", limit_item: "nofile", value: "1048576" }
```

The ansible-lint error:
```
WARNING  Listing 1 violation(s) that are fatal
args[module]: value of limit_type must be one of: soft, hard, -, got: {{ item.limit_type }} (warning)
test.yml:2 Task/Handler: Add or modify pam limits

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - experimental  # all rules tagged as experimental

                  Rule Violation Summary                   
 count tag          profile rule associated tags           
     1 args[module]         syntax, experimental (warning) 

```

##### Desired Behavior

Variable should be either ignored or the content of them should be parsed (but this is impossible in some cases). 

In the current stay the only solution is to blacklist the test globally.

**Comment by in-in** [comment]
In my case, I use the Default filter and expect the check to take the default value, but instead I get a warning.
```sh
args[module]: value of scope must be one of: file, local, global, system, got: {{ item.scope | default("global") }} (warning)
```

```yaml
- name: 'Git config'
  community.general.git_config:
    name: '{{ item.name }}'
    value: '{{ item.value }}'
    scope: '{{ item.scope | default("global") }}'
    repo: '{{ item.repo | default(None) }}'
  loop: '{{ git_config_data }}'
```

For the moment I skipped this check for this task with the `skip_ansible_lint` tag.

**Comment by timo-fc** [comment]
This is a good point default values in the template could be checked and setting them is not a problem.
